sql/delete.go

```diff
@@ -163,7 +163,7 @@ func canDeleteWithoutScan(n *parser.Delete, rows planNode, indexCount int) bool
 			}
 			return false
 		}
-		if n.Where != nil {
+		if scan := sel.table.node.(*scanNode); scan.filter != nil {
```

Comment:
LGTM, thanks for the fix!  Please add a comment here explaining that because there are no indexes, `table.node` cannot be an `indexJoin` node.